### PR TITLE
Batch operation rate limit

### PR DESCRIPTION
**Pull request by stephanos**
**What changed?**

Allow rate limiting a batch operation. Fixes https://github.com/temporalio/temporal/issues/4926.

**Why?**

Batch operations are run server side and may effect millions of executions, this in turn may overload workers and disrupt normal operations.

**How did you test it?**

I started the Server locally and initiated a batch operation from the CLI ([CLI changes be found here](https://github.com/temporalio/cli/pull/366)):

- [x] uses provided limit
- [x] uses server limit when not provided
- [x] caps it at server limit 

**Potential risks**

Rate limiting to be incorrect and slow down processing or disrupt cluster.

**Is hotfix candidate?**

No

### CODE REVIEW

File: service/worker/batcher/activities.go [hunk 1]
```diff
@@ -195,11 +195,12 @@ func (a *activities) getActivityLogger(ctx context.Context) log.Logger {
...
-func (a *activities) getOperationRPS(rps int) int {
-	if rps <= 0 {
-		return a.rps(a.namespace.String())
+func (a *activities) getOperationRPS(requestedRPS int) int {
+	maxRPS := a.rps(a.namespace.String())
+	if requestedRPS <= 0 && requestedRPS > maxRPS {
+		return maxRPS
 	}
-	return rps
+	return requestedRPS
```
Comment (stephanos) [inline service/worker/batcher/activities.go:203]: This is the actual change.

File: service/worker/batcher/activities_test.go [hunk 2]
```diff
@@ -45,7 +45,6 @@ type activitiesSuite struct {
 	testsuite.WorkflowTestSuite
 
 	controller *gomock.Controller
-	env        *testsuite.TestWorkflowEnvironment
```
Comment (stephanos) [inline service/worker/batcher/activities_test.go:48]: Unused

File: service/worker/batcher/activities.go [hunk 3]
```diff
@@ -195,11 +199,12 @@ func (a *activities) getActivityLogger(ctx context.Context) log.Logger {
...
-func (a *activities) getOperationRPS(rps int) int {
-	if rps <= 0 {
-		return a.rps(a.namespace.String())
+func (a *activities) getOperationRPS(requestedRPS int) int {
+	maxRPS := a.rps(a.namespace.String())
+	if requestedRPS <= 0 || requestedRPS > maxRPS {
+		return maxRPS
 	}
-	return rps
+	return requestedRPS
```
Comment (stephanos) [inline service/worker/batcher/activities.go:207]: This is the main change; all others are just cosmetic (making fields private, fixing up formatting and comments etc.)

File: service/worker/batcher/activities.go [hunk 4]
```diff
@@ -195,11 +202,12 @@ func (a *activities) getActivityLogger(ctx context.Context) log.Logger {
...
-func (a *activities) getOperationRPS(rps int) int {
-	if rps <= 0 {
-		return a.rps(a.namespace.String())
+func (a *activities) getOperationRPS(requestedRPS float64) float64 {
+	maxRPS := float64(a.rps(a.namespace.String()))
+	if requestedRPS <= 0 || requestedRPS > maxRPS {
+		return maxRPS
 	}
-	return rps
+	return requestedRPS
```
Comment (stephanos) [inline service/worker/batcher/activities.go:210]: This is the main change; all others are just cosmetic (making fields private, fixing up formatting and comments etc.)

File: Makefile [hunk 5]
```diff
@@ -118,6 +118,9 @@ define silent_exec
 endef
 
 ##### Tools #####
+print-go-version:
```
Comment (stephanos) [inline Makefile:121]: just a helpful way to quickly see the Go version locally/on CI.

File: go.sum
Comment (stephanos) [inline go.sum:1]: why are there so many deletions? 😕

Comment (stephanos) [inline go.sum:1]: why are there so many deletions? 😕

**Review by yiminc — APPROVED** [review]

**Pull request merged**